### ISSUE TITLE
NVDA does not read the contents of cells after cut

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#6397]
To identify the problem do the following.
Open a spreadsheet in Excel.
enter the cell a1 a date in the case typed 21/09/2016.
press ctrl + x on the cell a1.
Go to cell B1 and press ctrl + v to paste the cut cell A1.
It is expected that NVDA read the contents of the cell.
What happens is that NVDA no information, it's as if the cell b1 was empty.
From what I identified this occurs because the cell b1 is very small so that the content fully get the view.
However NVDA should have informed that the cell is overflowing with content and should read the contents of the cell.
To solve the problem it is necessary to self adjust the width of the cell.

using NVDA next 13593 and excel Microsoft Office 2013 version 15.0.4420.1017

**Comment by ehollig** [comment]
@fernando-jose-silva, is this still happening for you? I was able to perform the above steps without a problem in Excel 2016. What version of Excel are you using? Closing as worksforme, but if anyone is able to reproduce using a different version of Excel, it can be reopened.

**Comment by bhavyashah** [comment]
Using Excel 2010 and Windows 8.1, I am able to reproduce something strange and similar with the following STR:
* Type a date, for instance "21/01/16", in A1.
* Press Enter or Tab out of A1.
* Arrow back to A1.
* Given that you are not editing the cell and are simply focused on it, press Ctrl+X.
* Paste the clipboard contents in any other cell.
Expected Result: While navigating around the spreadsheet, NVDA should report 21/01/16 followed by the cell name when focus lands on the cell in which the cut content has been pasted.
Actual Result: NVDA reports "########" followed by the cell name, where the number of #s probably depends on the characters of the cut content. Strangely enough, pressing F2 to explore the same cell reports the original cut content accurately.

**Comment by fernando-jose-silva** [comment]
I can reproduce the problem even with Excel 2016 and 2013.
By putting the level of scoring for everyone I get the same results as Bhavyashah commented.
I pasted a nvda log where it shows the speech of the # characters.
I believe that as the contents of the cell are not fully visible which I believe is the reason why the cell contents are not read, nvda should pronounce something as non-visible.

IO - inputCore.InputManager.executeGesture (18:41:52.400):
Input: kb(laptop):rightArrow
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (18:41:52.454):
accRole failed: (-2147024809, 'Par\xe2metro incorreto.', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (18:41:52.460):
accRole failed: (-2147024809, 'Par\xe2metro incorreto.', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (18:41:52.464):
accRole failed: (-2147024809, 'Par\xe2metro incorreto.', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (18:41:52.469):
accRole failed: (-2147024809, 'Par\xe2metro incorreto.', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (18:41:52.471):
accRole failed: (-2147024809, 'Par\xe2metro incorreto.', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (18:41:52.480):
accRole failed: (-2147024809, 'Par\xe2metro incorreto.', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (18:41:52.490):
accRole failed: (-2147024809, 'Par\xe2metro incorreto.', (None, None, None, 0, None))
DEBUGWARNING - NVDAObjects.IAccessible.IAccessible._get_IAccessibleRole (18:41:52.500):
accRole failed: (-2147024809, 'Par\xe2metro incorreto.', (None, None, None, 0, None))
IO - speech.speak (18:41:52.618):
Speaking [u'########  B1']
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (18:41:52.625):
oleacc.AccessibleObjectFromEvent with window 67992, objectID 341 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (18:41:52.625):
oleacc.AccessibleObjectFromEvent with window 67992, objectID 342 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (18:41:52.638):
oleacc.AccessibleObjectFromEvent with window 67992, objectID 344 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (18:41:52.638):
oleacc.AccessibleObjectFromEvent with window 67992, objectID 345 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (18:41:52.640):
oleacc.AccessibleObjectFromEvent with window 67992, objectID 346 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (18:41:52.641):
oleacc.AccessibleObjectFromEvent with window 67992, objectID 347 and childID 0: [Error -2147467259] Erro não especificado
IO - inputCore.InputManager.executeGesture (18:41:55.200):
Input: kb(laptop):NVDA+f1
DEBUGWARNING - watchdog._watcher (18:41:57.207):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 199, in <module>
  File "core.pyc", line 396, in main
  File "wx\_core.pyc", line 8657, in MainLoop
  File "wx\_core.pyc", line 7952, in MainLoop
  File "core.pyc", line 367, in Notify
  File "queueHandler.pyc", line 83, in pumpAll
  File "queueHandler.pyc", line 50, in flushQueue
  File "scriptHandler.pyc", line 145, in _queueScriptCallback
  File "scriptHandler.pyc", line 187, in executeScript
  File "globalCommands.pyc", line 1420, in script_navigatorObject_devInfo
  File "logging\__init__.pyc", line 1167, in info
  File "logHandler.pyc", line 132, in _log
  File "wx\_core.pyc", line 13263, in SetInsertionPointEnd

INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (18:41:57.457):
Developer info for navigator object:
name: u'########'
role: ROLE_TABLECELL
states: 
isFocusable: False
hasFocus: False
Python object: <NVDAObjects.window.excel.ExcelCell object at 0x0B0D5E10>
Python class mro: (<class 'NVDAObjects.window.excel.ExcelCell'>, <class 'NVDAObjects.window.excel.ExcelBase'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: None
location: (1, 205, 1364, 500)
value: ''
appModule: <'excel' (appName u'excel', process ID 12456) at address aec2bb0>
appModule.productName: u'Microsoft Office 2016'
appModule.productVersion: u'16.0.7766.2099'
TextInfo: <class 'NVDAObjects.window.excel.ExcelCellTextInfo'>
windowHandle: 67992
windowClassName: u'EXCEL7'
windowControlID: 0
windowStyle: 1460338688
windowThreadID: 10656
windowText: u'Pasta1'
displayText: u''

**Comment by ehollig** [comment]
Hmmm, @michaelDCurran, @jcsteh, thoughts from the log provided?

**Comment by Adriani90** [comment]
Actually there is a lot of Errors possible in Excel where you have the number sign shown instead of Content. this is also the same behavior for sighted People. You can find explanations for each error in MS Excel. For this particular one for example here:
https://support.office.com/en-us/article/how-to-correct-a-error-bf801d0a-2a6e-44bd-a70e-0f780ae8f11e

For many other Errors for example here:
https://www.excel-easy.com/functions/formula-errors.html

I am closing this issue as works for me because it is a behavior defined in Excel itself and it is at the user of Excel to learn this Errors and how to solve them.
